clients/src/test/java/org/apache/kafka/common/security/oauthbearer/OAuthBearerLoginModuleTest.java

```diff
@@ -271,38 +374,113 @@ public void login1Commit1Login2Abort2Login3Commit3Logout3() throws LoginExceptio
         assertEquals(0, privateCredentials.size());
+        assertEquals(0, publicCredentials.size());
         loginModule1.commit();
         // Now we should have the first token
         assertEquals(1, privateCredentials.size());
+        publicCredentials = subject.getPublicCredentials(SaslExtensions.class);
+        assertEquals(1, publicCredentials.size());
         assertSame(tokens[0], privateCredentials.iterator().next());
+        assertSame(extensions[0], publicCredentials.iterator().next());
 
         // Now go get the second token
         loginModule2.login();
         // Should still have first token
         assertEquals(1, privateCredentials.size());
+        publicCredentials = subject.getPublicCredentials(SaslExtensions.class);
+        assertEquals(1, publicCredentials.size());
         assertSame(tokens[0], privateCredentials.iterator().next());
+        assertSame(extensions[0], publicCredentials.iterator().next());
         loginModule2.abort();
         // Should still have just the first token because we aborted
         assertEquals(1, privateCredentials.size());
         assertSame(tokens[0], privateCredentials.iterator().next());
+        publicCredentials = subject.getPublicCredentials(SaslExtensions.class);
+        assertEquals(1, publicCredentials.size());
+        assertSame(extensions[0], publicCredentials.iterator().next());
 
         // Now go get the third token
         loginModule2.login();
         // Should still have first token
         assertEquals(1, privateCredentials.size());
         assertSame(tokens[0], privateCredentials.iterator().next());
+        publicCredentials = subject.getPublicCredentials(SaslExtensions.class);
+        assertEquals(1, publicCredentials.size());
+        assertSame(extensions[0], publicCredentials.iterator().next());
         loginModule2.commit();
         // Should have first and third tokens at this point
         assertEquals(2, privateCredentials.size());
         Iterator<Object> iterator = privateCredentials.iterator();
         assertNotSame(tokens[1], iterator.next());
         assertNotSame(tokens[1], iterator.next());
+        publicCredentials = subject.getPublicCredentials(SaslExtensions.class);
+        assertEquals(2, publicCredentials.size());
+        Iterator<SaslExtensions> publicIterator = publicCredentials.iterator();
+        assertNotSame(extensions[1], publicIterator.next());
+        assertNotSame(extensions[1], publicIterator.next());
         loginModule1.logout();
         // Now we should have just the third token
         assertEquals(1, privateCredentials.size());
         assertSame(tokens[2], privateCredentials.iterator().next());
+        publicCredentials = subject.getPublicCredentials(SaslExtensions.class);
+        assertEquals(1, publicCredentials.size());
+        assertSame(extensions[2], publicCredentials.iterator().next());
+    }
+
+    @Test
+    public void commitPopulatesExtensions() throws LoginException {
```

Comment:
I believe this test adds no value and should be eliminated because the case is covered above.

Reply:
Take a look at my comment below for test `commitDoesNotThrowOnUnsupportedExtensionsCallback`. I do not feel as strongly about this test as I feel on the one below, but I also tentatively think it doesn't hurt to have one more test

Reply:
Yes, let's eliminate this one and keep the one below.